### PR TITLE
rewrite docs and use github pages

### LINKED ISSUE #322: Can you provide a XXX gradle. KTS use case, thank you very much

### LINKED ISSUE #332: AGP 7.1 supports javadocJar, sourcesJar and variant publishing
Hi, I just read these 3 release notes, I thought I would call it out:
https://developer.android.com/studio/releases/gradle-plugin#build-variant-publishing

Does this mean most of this plugin can be replaced with standard Gradle publishing / AGP configuration?

See also https://developer.android.com/studio/build/maven-publish-plugin

### LINKED ISSUE #426: v0.22.0 is automatically closing non-staging artifacts
Call me old fashioned but I prefer to manually close & release artifacts... as a result, I discovered the latest version (0.22.0) automatically closes when you call `./gradlew publish --no-daemon --no-parallel`.

The logs seem to indicate that it is trying to create & close the *staging* repository, but in reality it's closing the *prod* repository:

```
> Task :library:createStagingRepository
Creating repository in profile: net.danlew
Created staging repository netdanlew-1065
Closing repository: netdanlew-1065
⠴ waiting for close...                                               Repository netdanlew-1065 closed
```

Here's what I opened OSS sonatype to see:

image

### PR DESCRIPTION
**Pull request by gabrielittner**
Rewrites the docs completely. I've split it into 2 separate guides one for maven central and then one for other targets. The base plugin also gets it's own subpage. For everything we show the groovy, the kts and the gradle.properties version.

I left a few todos where I know that I still need to do something but this can already be reviewed. Hopefully I didn't forget any parts that we had before.

To try this out locally 
```
brew install mkdocs
pip3 install mkdocs-material
cp README.md docs/index.md
mkdocs serve
```

@vanniktech If you enable github pages for this repo I could also change the workflow to for now publish from this branch so that we have a working Github pages version to look at already

Closes #426, closes #322, closes #332, closes #427, closes #264

### CODE REVIEW

**Comment by vanniktech** [comment]
Like this?

Screen Shot 2022-10-12 at 09 33 48

**Comment by vanniktech** [comment]
So like this?

Screen Shot 2022-10-12 at 22 16 46

**Review by vanniktech — APPROVED** [review]
Really good stuff!

**Comment by gabrielittner** [comment]
Pushed the missing parts, https://vanniktech.github.io/gradle-maven-publish-plugin should now have everything

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Publish Docs
+
+on:
+  push:
+    branches:
+      - docs
```
Comment (gabrielittner) [inline .github/workflows/publish-docs.yml:6]: Before merging:
```suggestion
      - master
```

**Review by vanniktech — APPROVED** [review]
🙏

**Pull request merged**